ros_cross_compile/data_collector.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for time series data collection and writing said data to a file."""
+
+from contextlib import contextmanager
+from datetime import datetime
+from enum import Enum
+import json
+from pathlib import Path
+import time
+from typing import NamedTuple, Union
+
+from ros_cross_compile.sysroot_creator import INTERNALS_DIR
+
+
+Datum = NamedTuple('Datum', [('name', str),
+                             ('value', Union[int, float]),
+                             ('unit', str),
+                             ('timestamp', float),
+                             ('complete', bool)])
+
+
+class Units(Enum):
+    time = 'seconds'
+    size = 'megabytes'
+
+
+class DataCollector:
+    """Provides an interface to collect time series data."""
+
+    def __init__(self):
+        self.data = {}
```

Comment:
I don't think this needs to be a dictionary, based on below comment about names. This can just be a list of `Datum`s